cl/merkle_tree/hasher.go

```diff
@@ -83,6 +87,14 @@ func (m *merkleHasher) getBuffer(size int) [][32]byte {
 	return m.internalBuffer[:size]
 }
 
+// getBuffer provides buffer of given size.
+func (m *merkleHasher) getBufferForSSZList(size int) [][32]byte {
+	if size > len(m.internalBufferForSSZList) {
+		m.internalBufferForSSZList = make([][32]byte, size*2)
```

Comment:
Why multiply the size by 2 and not set it as size?

Reply:
we make sure to allocate some memory if we increase by one.

so if we append on a slice and we don't have  anything pre-allocated what happens is that you make new slice and append 2x. like this we do it ourselves but without cap only len